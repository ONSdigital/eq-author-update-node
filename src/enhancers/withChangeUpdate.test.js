import React from "react";
import { shallow } from "enzyme";

import withChangeUpdate from "./withChangeUpdate";

const EnhancedComponent = withChangeUpdate("div");

const createWrapper = (props, render = shallow) =>
  render(<EnhancedComponent {...props} />);

describe("withChangeUpdate", () => {
  let props, wrapper;
  let onChange = jest.fn();
  let onUpdate = jest.fn();

  beforeEach(() => {
    props = {
      onChange: onChange,
      onUpdate: onUpdate,
    };

    wrapper = createWrapper(props);
  });

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should correctly handle change update", () => {
    const update = { foo: "bar" };
    wrapper.simulate("changeUpdate", update);
    expect(onChange).toHaveBeenCalledWith(update, onUpdate);
  });
});
