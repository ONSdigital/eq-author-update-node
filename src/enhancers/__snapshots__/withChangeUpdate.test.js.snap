// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withChangeUpdate should render 1`] = `
<div
  onChange={[MockFunction]}
  onChangeUpdate={[Function]}
  onUpdate={[MockFunction]}
/>
`;
